api/clients/eigenda_client.go

```diff
@@ -41,7 +42,13 @@ func NewEigenDAClient(log log.Logger, config EigenDAClientConfig) (*EigenDAClien
 		return nil, fmt.Errorf("failed to parse EigenDA RPC: %w", err)
 	}
 
-	signer := auth.NewLocalBlobRequestSigner(config.SignerPrivateKeyHex)
+	var signer core.BlobRequestSigner
+	if len(config.SignerPrivateKeyHex) == 64 {
+		signer = auth.NewLocalBlobRequestSigner(config.SignerPrivateKeyHex)
+	} else {
```

Comment:
only do this when `config.SignerPrivateKeyHex` is nil, and return error when `config.SignerPrivateKeyHex` is anything other than nil or has length != 64

Reply:
I can add the error case, but `config.SignerPrivateKeyHex` is a string, what I am missing here, there is no reason to check nil

Reply:
I want to make sure we check the key length. If this key is provided (non-empty string), it should be length 64 and error out otherwise.

Reply:
yes, pushed that change